src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -304,1237 +234,653 @@ export default class Fms {
-    }
+    // /**
+    // * Build a `LegModel` instance that contains a `WaypointModel` with hold properties
+    // *
+    // * @for Fms
+    // * @method _createLegWithHoldWaypoint
+    // * @param waypointProps {object}
+    // * @return legModel {LegModel}
+    // */
+    // _createLegWithHoldWaypoint(waypointProps) {
+    //     const legModel = new LegModel(
+    //         waypointProps.name,
+    //         this.currentRunwayName,
+    //         this.currentPhase,
+    //         this._navigationLibrary,
+    //         waypointProps
+    //     );
+    //
+    //     return legModel;
+    // }
 
     /**
-     * Regenerate SID leg based on current runway assignment
+     * Update the expected arrival runway based on the STAR's exit point runway
      *
      * @for Fms
-     * @method _regenerateSidLeg
+     * @method _updateArrivalRunwayFromRoute
      */
-    _regenerateSidLeg() {
-        const sidLegIndex = this._findLegIndexForProcedureType(PROCEDURE_TYPE.SID);
-
-        if (sidLegIndex === INVALID_INDEX) {
-            // would be throwing here, except that causes the initial setting of the
-            // departure runway to throw. So instead will just return early.
-            return;
-        }
-
-        const sidLeg = this.legCollection[sidLegIndex];
-
-        this.replaceDepartureProcedure(sidLeg.routeString, this.departureRunwayModel);
-    }
+    _updateArrivalRunwayFromRoute() {
+        const arrivalRunwayModel = this._routeModel.getArrivalRunwayModel();
 
-    /**
-     * Regenerate STAR leg based on current runway assignment
-     *
-     * @for Fms
-     * @method _regenerateStarLeg
-     */
-    _regenerateStarLeg() {
-        const starLegIndex = this._findLegIndexForProcedureType(PROCEDURE_TYPE.STAR);
+        if (_isNil(arrivalRunwayModel)) {
+            console.error('Expected route to have a valid arrival runway');
 
-        if (starLegIndex === INVALID_INDEX) {
-            // would be throwing here, except that causes the initial setting of the
-            // arrival runway to throw. So instead will just return early.
             return;
         }
 
-        const starLeg = this.legCollection[starLegIndex];
-
-        this.replaceArrivalProcedure(starLeg.routeString, this.arrivalRunwayModel);
+        this.setArrivalRunway(arrivalRunwayModel);
     }
 
     /**
-     * Set the currentPhase with the appropriate value, based on the spawn category
+     * Update the expected departure runway based on the SID's entry point runway
      *
      * @for Fms
-     * @method _setCurrentPhaseFromCategory
-     * @param category {string}
-     * @private
+     * @method _updateArrivalRunwayFromRoute
      */
-    _setCurrentPhaseFromCategory(category) {
-        switch (category) {
-            case FLIGHT_CATEGORY.ARRIVAL:
-                this.setFlightPhase(FLIGHT_PHASE.CRUISE);
+    _updateDepartureRunwayFromRoute() {
+        const departureRunwayModel = this._routeModel.getDepartureRunwayModel();
```

Comment:
without having seen the `RouteModel`, I feel like this should be a simple getter.